### PR TITLE
Update generated key with latest timestamp when index is changed

### PR DESCRIPTION
**Pull request by josephprem**

### CODE REVIEW

File: go.sum
Comment (hypnoglow) [inline go.sum:1]: Deleted by accident? We want to keep this.

File: internal/helmutil/index_v2.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ func (idx *IndexV2) SortEntries() {
 	idx.index.SortEntries()
 }
 
+func (idx *IndexV2) TimeStamp() {
```
Comment (hypnoglow) [inline internal/helmutil/index_v2.go:124]: Perhaps we should clarify the name?

```suggestion
func (idx *IndexV2) UpdateGeneratedTime() {
```

File: plugin.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: "s3"
-version: "0.15.1"
```
Comment (hypnoglow) [inline plugin.yaml:2]: Please don't update the version here. It should be done separately when the release is prepared.

**Review by hypnoglow — CHANGES_REQUESTED** [review]
Thank you! I propose a few changes, everything else looks good.

File: go.sum
Comment (josephprem) [reply go.sum:1]: Sorry, It was a mistake

File: internal/helmutil/index_v2.go (hunk 1 above)
Comment (josephprem) [reply internal/helmutil/index_v2.go:124]: Agree

File: plugin.yaml (hunk 2 above)
Comment (josephprem) [reply plugin.yaml:2]: Ok

File: internal/helmutil/index_v2_test.go [hunk 3]
```diff
@@ -122,3 +122,16 @@ func TestIndexV2_AddOrReplace(t *testing.T) {
 		assert.Equal(t, "sha256:222", i.index.Entries["foo"][0].Digest)
 	})
 }
+
+func TestIndexV2_Timestamp(t *testing.T) {
```
Comment (hypnoglow) [inline internal/helmutil/index_v2_test.go:126]:
```suggestion
func TestIndexV2_UpdateGeneratedTime(t *testing.T) {
```

File: internal/helmutil/index_v2.go [hunk 4]
```diff
@@ -121,6 +121,10 @@ func (idx *IndexV2) SortEntries() {
 	idx.index.SortEntries()
 }
 
+func (idx *IndexV2) UpdateGeneratedTime() {
+	idx.index.Generated = time.Now()
```
Comment (hypnoglow) [inline internal/helmutil/index_v2.go:125]:
```suggestion
	idx.index.Generated = time.Now().UTC()
```

Let's do this so we're not relying on OS timezone.

File: internal/helmutil/index_v3.go [hunk 5]
```diff
@@ -123,6 +123,10 @@ func (idx *IndexV3) SortEntries() {
 	idx.index.SortEntries()
 }
 
+func (idx *IndexV3) UpdateGeneratedTime() {
+	idx.index.Generated = time.Now()
```
Comment (hypnoglow) [inline internal/helmutil/index_v3.go:127]:
```suggestion
	idx.index.Generated = time.Now().UTC()
```

File: internal/helmutil/index_v3_test.go [hunk 6]
```diff
@@ -122,3 +122,16 @@ func TestIndexV3_AddOrReplace(t *testing.T) {
 		assert.Equal(t, "sha256:222", i.index.Entries["foo"][0].Digest)
 	})
 }
+
+func TestIndexV3_Timestamp(t *testing.T) {
```
Comment (hypnoglow) [inline internal/helmutil/index_v3_test.go:126]:
```suggestion
func TestIndexV3_UpdateGeneratedTime(t *testing.T) {
```

File: cmd/helm-s3/push.go [hunk 7]
```diff
@@ -211,7 +211,10 @@ func (act *pushAction) run(ctx context.Context) error {
 		return errors.WithMessage(err, "add/replace chart in the index")
 	}
 	idx.SortEntries()
-
+	
+	// #142: update timestamp of the generated field	
+	idx.TimeStamp()
```
Comment (hypnoglow) [inline cmd/helm-s3/push.go:216]:
```suggestion
	idx.UpdateGeneratedTime()
```

**Review by hypnoglow — CHANGES_REQUESTED** [review]

**Comment by hypnoglow** [comment]
Fixes #142

**Review by hypnoglow — APPROVED** [review]

**Comment by hypnoglow** [comment]
Hey, I made a few changes I proposed previously. Thank you again for the contribution.

**Pull request merged**